### PR TITLE
Add info to countDocuments spec - needs review

### PR DESCRIPTION
**Pull request by johnsmartco**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: docs/jsonapi-spec.md [hunk 1]
```diff
@@ -473,10 +473,38 @@ Commands are defined using the BNF-like syntax, with samples presented using a [
 
 ### countDocuments Command
 
+Returns the count of documents that match the query for a collection or view.
+
+*Sample:*
+
+```json
+db.orders.countDocuments({})
```
Comment (maheshrajamani) [inline docs/jsonapi-spec.md:481]: The json needs to be like `{
                        "countDocuments": {
                            "filter": {"location": "London", "race.competitors" : {"$eq" : 100}}
                        }
                      }`

File: docs/jsonapi-spec.md [hunk 2]
```diff
@@ -473,10 +473,38 @@ Commands are defined using the BNF-like syntax, with samples presented using a [
...
+See the IMPORTANT note in https://www.mongodb.com/docs/manual/reference/method/db.collection.countDocuments/#transactions.
+
+* Jeff: Should we also ask core DB for estimated count?
+
+#### Count all documents that match a query
+
+Count the number of the documents in the `purchase` collection with the field `order_date` greater than new Date('01/01/2022'):
+
+```json5
+db.purchase.countDocuments( { order_date: { $gt: new Date('01/01/2022') } }, { limit: 100 } )
```
Comment (maheshrajamani) [inline docs/jsonapi-spec.md:505]: Same as previous comment the json request structure need to be fixed. 
There are no options for this command. Need to remove the  { limit: 100 }

File: docs/jsonapi-spec.md [hunk 3]
```diff
@@ -473,10 +473,38 @@ Commands are defined using the BNF-like syntax, with samples presented using a [
...
 
+#### countDocuments TODO considersations
+
+re: https://datastax.jira.com/browse/C2-2491
+
+* What are the limitations on the filters?
+
+* What are the performance problems?
+
+See the IMPORTANT note in https://www.mongodb.com/docs/manual/reference/method/db.collection.countDocuments/#transactions.
```
Comment (maheshrajamani) [inline docs/jsonapi-spec.md:496]: These comment lines need to be removed, not sure if needs to be in spec. May be start a new discussion or issue for these comment in github.

File: docs/jsonapi-spec.md [hunk 4]
```diff
@@ -473,10 +473,38 @@ Commands are defined using the BNF-like syntax, with samples presented using a [
...
+
+* Jeff: Should we also ask core DB for estimated count?
+
+#### Count all documents that match a query
+
+Count the number of the documents in the `purchase` collection with the field `order_date` greater than new Date('01/01/2022'):
```
Comment (maheshrajamani) [inline docs/jsonapi-spec.md:502]: The collection name is part of url, so we can avoid this. 
Currently working on date type support in JsonAPI, so the filter will be like below. Better to use existing filter like something I have suggested.

Example date filter:
`"filter" : { "order_date"  : {"$date" :  1672531200000 }}`. -> this is in ejson format and epoch (1672531200000) represent 1/1/2023 00:00:00 UTC.

**Review by maheshrajamani — CHANGES_REQUESTED** [review]

File: docs/jsonapi-spec.md (hunk 1 above)
Comment (johnsmartco) [reply docs/jsonapi-spec.md:481]: Thanks!

File: docs/jsonapi-spec.md (hunk 2 above)
Comment (johnsmartco) [reply docs/jsonapi-spec.md:505]: thanks

File: docs/jsonapi-spec.md (hunk 3 above)
Comment (johnsmartco) [reply docs/jsonapi-spec.md:496]: done in next commit

File: docs/jsonapi-spec.md (hunk 4 above)
Comment (johnsmartco) [reply docs/jsonapi-spec.md]: Done in latest commit.

**Comment by johnsmartco** [comment]
latest commit cleaned up countDocuments example json

**Review by maheshrajamani — APPROVED** [review]
LGTM

**Pull request merged**